### PR TITLE
Align new prisoner search response with old endpoint

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java [hunk 1]
```diff
@@ -535,7 +535,7 @@ public List<OffenderPropertyContainer> getActivePropertyContainers() {
 
     public List<OffenderProfileDetail> getActiveProfileDetails() {
         return profileDetails.stream()
-            .filter(pd -> Objects.nonNull(pd.getCode()))
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:538]: If a profile value is a number then it's not a PROFILE_CODE so this is null and the value is stored in propert profileCode

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerSearchService.kt [hunk 2]
```diff
@@ -30,7 +30,7 @@ class PrisonerSearchService(
           lastName = it.lastName,
           dateOfBirth = it.dateOfBirth,
           agencyId = it.agencyId,
-          alerts = it.alerts,
+          alerts = it.alerts.sortedBy { it.alertId },
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerSearchService.kt:33]: Retain the order from the old endpoint

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerSearchService.kt [hunk 3]
```diff
@@ -40,7 +40,7 @@ class PrisonerSearchService(
           csra = it.csra,
           categoryCode = it.categoryCode,
           inOutStatus = it.inOutStatus,
-          identifiers = it.identifiers,
+          identifiers = it.identifiers.sortedBy { it.whenCreated },
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerSearchService.kt:43]: Retain the order from the old endpoint

File: src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderTransformer.java [hunk 4]
```diff
@@ -62,11 +64,11 @@ public InmateDetail transform(final OffenderBooking latestBooking) {
             .sentenceTerms(sentenceTerms)
             .sentenceDetail(latestBooking.getSentenceCalcDates())
             .profileInformation(latestBooking.getActiveProfileDetails().stream()
-                .filter(pd -> pd.getCode() != null)
+                .sorted(Comparator.comparing(OffenderProfileDetail::getListSequence))
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderTransformer.java:67]: Retain the order from the old endpoint

**Review by petergphillips — APPROVED** [review]
LGTM

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**